openpype/hosts/maya/api/lib.py

```diff
@@ -2565,6 +2565,66 @@ def bake_to_world_space(nodes,
 
     """
+    @contextlib.contextmanager
+    def _revert_lock_attributes(node, new_node):
+        # Connect all attributes on the node except for transform
+        # attributes
+        attrs = _get_attrs(node)
+        attrs = set(attrs) - transform_attrs if attrs else []
+        original_attrs_lock = {}
+        try:
+            for attr in attrs:
+                orig_node_attr = '{0}.{1}'.format(node, attr)
+                new_node_attr = '{0}.{1}'.format(new_node, attr)
+                original_attrs_lock[new_node_attr] = (
+                    cmds.getAttr(new_node_attr, lock=True)
+                )
+
+                # unlock to avoid connection errors
+                cmds.setAttr(new_node_attr, lock=False)
+
+                cmds.connectAttr(orig_node_attr,
+                                 new_node_attr,
+                                 force=True)
+            yield
+        finally:
+            for attr, lock_state in original_attrs_lock.items():
+                cmds.setAttr(attr, lock=lock_state)
+
+    @contextlib.contextmanager
+    def _revert_lock_shape_attributes(node, new_node, shape):
+       # If shapes are also baked then connect those keyable attributes
+        if shape:
+            children_shapes = cmds.listRelatives(new_node,
+                                                 children=True,
+                                                 fullPath=True,
+                                                 shapes=True)
+            if children_shapes:
+                orig_children_shapes = cmds.listRelatives(node,
+                                                          children=True,
+                                                          fullPath=True,
+                                                          shapes=True)
+                original_shape_lock_attrs = {}
+                try:
+                    for orig_shape, new_shape in zip(orig_children_shapes,
+                                                     children_shapes):
+                        attrs = _get_attrs(orig_shape)
+                        for attr in attrs:
+                            orig_node_attr = '{0}.{1}'.format(orig_shape, attr)
+                            new_node_attr = '{0}.{1}'.format(new_shape, attr)
+                            original_shape_lock_attrs[new_node_attr] = (
+                                cmds.getAttr(new_node_attr, lock=True)
+                            )
+                            # unlock to avoid connection errors
+                            cmds.setAttr(new_node_attr, lock=False)
+
+                            cmds.connectAttr(orig_node_attr,
+                                             new_node_attr,
+                                             force=True)
+                    yield
+                finally:
+                    for attr, lock_state in original_shape_lock_attrs.items():
+                        cmds.setAttr(attr, lock=lock_state)
```

Comment:
Honestly, this is way more complexity then needed.

This should just be a context manager that "unlocks" it.

```python
@contextlib.contetmanager
def unlock(attrs):
    """Unlock attributes during context"""
    locked = []
    try:
        for attr in attrs:
            if cmds.getAttr(attr, lock=True):
                locked.append(attr)
                cmds.setAttr(attr, lock=False)
        yield
    finally:
        for attr in locked:
            cmds.setAttr(attr, lock=True)
```

The rest of the logic needs no context managing - but only connecting to the unlocked attributes.
Can we clean this up so that the context manager is simple?


In essence it's just:

- Duplicate
- Unlock attributes on duplicate
- Connect original attribute to the duplicated one
- Bake

Which should somehow turn out to be roughly as simple as:

```
new_node = duplicate(node)
attrs = _get_attrs(new_nodes)
with unlocked(attrs):
    connect(node, new_node, attrs)
    bake(new_node)
```

Only downside is that we're also dealing with the shapes so also need to do similar for the shape.

Reply:
this is a good start for us but I find out that this way can't really revert the settings for this particular function are connecting the animation curves nodes (from some attributes of original cameras) to baked cameras. 
maybe the function with the contextlib needs to get the locked attributes original camera so that it can revert it.

Reply:
No idea what you're saying :D

Reply:
I have tried to introduce the similar contextlib.contextmanger functions similar to what you suggested, it can't still revert the lock status of the camera attributes after publishing.

Reply:
Do you need me to look into this?